support/metrics/metricsTracker.go

```diff
@@ -157,6 +186,20 @@ func (mt *MetricsTracker) SendStartupEvent() error {
 	return mt.sendEvent(startupEventName, mt.props)
 }
 
+// SendGuiEvent sends an event from the GUI.
+// TODO DS Determine generic properties to send with the GUI event.
+func (mt *MetricsTracker) SendGuiEvent(eventName string) error {
```

Comment:
when you have two factory methods -- makeCLi and makeGui and then two methods on it sendEvent and sendGuiEvent it's a sign that we need an interface and two separate implementations.

A good way to know this is if the developer needs to know that we cannot use sendEvent for gui based metrics -- that's a sign.

can we not reuse the `sendEvent` method and export that?

Reply:
if we have logic that we always send the `mt.props` and merge in any values passed in the `map[string]interface{}` into `mt.props` then we can achieve this -- i.e. the merging can happen in sendEvent as opposed to in `updateProps`

Reply:
How do you do this merge? I get what needs to be done, but I can't figure out a syntax that would work.

Reply:
Never mind, I've figured out an approach that generates correct output. I don't think there's another way to merge two arbitrary structs, and it works for our purposes. Let me know what you think.